neo-codap/src/graph/graph.tsx

```diff
@@ -86,14 +223,24 @@ export class GraphComponent extends React.Component<IGraphProps, IGraphState> {
             attrIDs.push(yAttrID);
             yAttr = srcDataSet.attrFromID(yAttrID);
         }
-    
+        if (!legendAttrID) {
+            legendAttrID = this.findLegendAttribute(srcDataSet);
+        }
+        if (legendAttrID && (legendAttrID !== 'none')) {
+            attrIDs.push(legendAttrID);
+            legendAttr = srcDataSet.attrFromID(legendAttrID);
+            legendAttrType = this.getLegendAttrType(srcDataSet, legendAttr);
+        }
+
+        const isEmpty = (value: string|number|undefined) => (value == null) || (value === '');
```

Comment:
Adding this function is a nice improvement. To me the declaration is more readable if it's
```
const isEmpty = (value?: string|number) => ...
```
which should be equivalent, but that's a matter of taste.

Reply:
This actually should be IAttribute which I found exported after I wrote this part of the code.  I'll update it.